lib/web/apiserver_test.go

```diff
@@ -1515,6 +1542,24 @@ func TestUnifiedResourcesGet(t *testing.T) {
 	}
 	require.Equal(t, expectedRoles, listResp.Items[0].AWSRoles)
 	t.Log(string(re.Bytes()), listResp)
+
+	// query only apps, SAMLIdPServiceProvider 'app' should be included
+	type appResponse struct {
+		Items      []webui.App `json:"items"`
+		TotalCount int         `json:"totalCount"`
+	}
+	query = url.Values{"sort": []string{"name"}, "kinds": []string{types.KindApp}, "query": []string{``}}
+	re, err = pack.clt.Get(context.Background(), endpoint, query)
+	require.NoError(t, err)
+	appRes := appResponse{}
+	require.NoError(t, json.Unmarshal(re.Bytes(), &appRes))
+	require.Len(t, appRes.Items, 3)
+	for _, app := range appRes.Items {
+		require.Equal(t, types.KindApp, app.Kind)
+	}
+	require.False(t, appRes.Items[0].SAMLApp)
+	require.False(t, appRes.Items[1].SAMLApp)
+	require.True(t, appRes.Items[2].SAMLApp)
```

Comment:
I am not fan of these kinds of test that just selectively test for one or two values. Instead, prefer to test the whole response object so it will help catch any regressions in the future when we add more changes in the core resource spec but forget to update UI response type or vice versa. 

You can use `MakeApp` and `MakeAppTypeFromSAMLApp` to convert the resources we created earlier in this test and then assert if elements matches in the response. 

That said, I also do not want to hold you on this and will let you decide :)